gc/base/Bits.hpp

```diff
@@ -43,6 +44,44 @@ class MM_Bits {
 		return x << OMR_LOG_POINTER_SIZE;
 	}
 
+	/**
+	 * log2(size of an object to object reference)
+	 *
+	 * @param[in] compressed true if object to object references are compressed, false if not
+	 * @return the shift value
+	 */
+	MMINLINE static uintptr_t logReferenceSize(bool compressed) { return compressed ? 2 : OMR_LOG_POINTER_SIZE; }
+
+	/**
+	 * Calculate the difference between two object slot pointers, in slots
+	 *
+	 * @param[in] p1 the value to be subtracted from
+	 * @param[in] p2 the value to be subtracted
+	 * @param[in] compressed true if object to object references are compressed, false if not
+	 * @return p1 - p2 in slots
+	 */
+	MMINLINE static uintptr_t subtractSlotPointers(fomrobject_t *p1, fomrobject_t *p2, bool compressed) { return ((uintptr_t)p1 - (uintptr_t)p2) >> logReferenceSize(compressed); }
```

Comment:
should we call it subtractReferences then (and if so, update the other two methods that you introduced)?

Reply:
I was thinking "object slot pointer" or "reference pointer" (I think it would be a good idea to keep pointer in there, `addIndexToReference` would be confusing, I think).

If we're going to make a change, I would suggest:
```
subtractFromReferencePointer
addToReferencePointer
subtractReferencePointers
```

Reply:
Hm, not sure why you think 'pointer' is helpful here. We have, for example, existing methods like: GC_SlotObject::readReferenceFromSlot()

But if we go with it, between the two you suggested I think I actually prefer 'object slot pointer'. 

Clearly, it's matter of perception and very subjective (hence leads to confusions), so let's see if @dmitripivkine @rwy0717 have opinions?

Reply:
I guess I was thinking that the fomrobject_t is the reference, and we're dealing with pointers to slots containing those (there, managed to use all 3 terms in the same sentence!).

I have no strong feelings here, so I'll go with the flow.

Reply:
Here is another suggestion... Move these new methods to GC_SlotObject which is clearly dealing with _object_ slots.  Hoping the class is visible from all sites you intend to use and cause some other type of confusion. 

This class has been implying native slots so far, and adding reference slot functionality is what confused me (and will possibly somebody else in future).

Reply:
That sounds reasonable - I'll see if I can make it work (i.e. if I'm able to include the header in all of the appropriate places without causing compile problems).

Reply:
Functions have been relocated to SlotObject (under the same names for now).

Reply:
Now that I'm looking the change again, definitely this is a better place.

As for the naming, I'm realizing 'reference' (instead of slotPointer) as I had suggested before would be completely wrong here (since in this context references are clearly addresses to beginning of an object).

But inspired by naming of writeAddressToSlot(), you can consider using slotAddress instead of slotPointer, but I'm ok either way.

Reply:
New function names:
```
subtractSlotAddresses
addToSlotAddress
subtracFromSlotAddress
```